src/main/java/org/littleshoot/proxy/extras/SelfSignedSslEngineSource.java

```diff
@@ -70,6 +98,83 @@ public SelfSignedSslEngineSource() {
     public SSLEngine newSslEngine() {
         return sslContext.createSSLEngine();
     }
+    
+    public SSLEngine clientSslEngineFor(SSLSession remoteServerSslSession) {
+    	try{
+
+    		String sslContextKey = remoteServerSslSession.getPeerHost() +":" + remoteServerSslSession.getPeerPort();
+    		SSLContext newSslContext = null;
+    		if(!sslContextCache.contains(sslContextKey) || (newSslContext = sslContextCache.get(sslContextKey).get()) == null ){
```

Comment:
Why the double check here?  Why not just get the newSslContext and compare that to null?